flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PartitionedFileWriter.java

```diff
@@ -205,32 +215,70 @@ public void writeBuffers(List<BufferWithChannel> bufferWithChannels) throws IOEx
             return;
         }
 
-        long expectedBytes = 0;
+        long expectedBytes;
         ByteBuffer[] bufferWithHeaders = new ByteBuffer[2 * bufferWithChannels.size()];
 
+        if (isBroadCastRegion) {
+            expectedBytes = writeBroadcastBuffers(bufferWithChannels, bufferWithHeaders);
```

Comment:
This two methods might need to change the names since it does not do the write. Perhaps change to something like `collectBroadcastBuffers` and `collectUnicastBuffers` ?